### ISSUE TITLE
Setup a docker node

### ISSUE THREAD

**Issue by Palak3196** [hashgraph/hedera-local-node#445]
Hi Team,

We're looking to start a hedera mainnet archival node (having data from block 0) without setting an account or pay any fee, so basically just a plain observer node. Is that possible? If yes could you please point me to the setup?

Thanks,
Palak

**Comment by Palak3196** [comment]
will a mirror node itself be sufficient for me to have all the data from block 1 to latest block? without paying any account fee etc?

**Comment by georgi-l95** [comment]
Yes, mirror-node has all the data from genesis block to latest. Also I believe without any account fees.

**Comment by Palak3196** [comment]
great, are these the steps to bring up the mirror node? (found them here: https://github.com/hashgraph/hedera-mirror-node)
git clone https://github.com/hashgraph/hedera-mirror-node.git
cd hedera-mirror-node
docker compose up

**Comment by georgi-l95** [comment]
> The charges associated with the downloading of stream files are paid for by the requester and not the bucket owner.

Could be, I'm not very familiar with the process. Will ask around and get back to you. 
Also keep in mind that this require large amount of space, you'd have to read and store more than 12 B transactions

**Comment by Palak3196** [comment]
Yes please.
I'm okay with any amount of space, CPU and RAM required. But my organisation doesn't allow paying for having nodes up.

Thanks,
Palak

**Comment by steven-sheehy** [comment]
You can't run a mirror node of any size without paying for S3 retrieval. If you want all data the costs would dramatically increase. The current database size is 48 TB and you can see that at Google Cloud SQL disks [costs](https://cloud.google.com/sql/pricing#storage-networking-prices) that would be quite costly. See the Hedera [docs](https://docs.hedera.com/hedera/core-concepts/mirror-nodes/run-your-own-beta-mirror-node) on how to run a mirror node.

**Comment by david-bakin-sl** [comment]
also remember your use case might not actually need _everything_ from the beginning of time.  the mirror node has configuration options where you get some flexibility to describe the information you want to keep in it - beyond just time-based constraints.

I'm not sure how you estimate _in advance_ how much space you'll need if you trim away the stuff you don't need to persist, and it does depend on your use case, but that approach can work.  

(however, you still need to _ingest_ all the record streams as all the different kinds of transactions and entities are of course "mixed up" in there in timestamp order ...)

**Comment by Palak3196** [comment]
Thanks Team, this clears up a lot of things.

My use case is: We need transactions for any given address, if there any direct api (http/grpc) that would be able to query my mirror node?
 
If the above is not possible, then from the node, we need to have transactions for each block using which we can get the transactions for an address by scanning the blocks.

**Comment by david-bakin-sl** [comment]
Would [this - `/api/v1/transactions`](https://mainnet-public.mirrornode.hedera.com/api/v1/docs/#/transactions/listTransactions) work for you? It has an `account.id` query parameter.  I think it's paginated too.
* And of course if your team needs an SLA guarantee there are commercial providers of mirror node services.

(BTW, the whole purpose of the mirror node is to answer queries like this.  If you find something you need it doesn't do then _that's_ something you can ask about!)
